cmake/Tools/Platform/Android/android_post_build.py

```diff
@@ -0,0 +1,256 @@
+    :param asset_mode:          The desired asset mode to determine the layout rules
+    :param asset_bundle_folder: (For PAK asset modes) the location of where the PAK files are expected.
+    """
+
+    android_app_root_path = Path(android_app_root)
+    if not android_app_root_path.is_dir():
+        raise AndroidPostBuildError(f"Invalid Android Gradle build path: {android_app_root_path} is not a directory or does not exist.")
+
+    if gradle_version < MINIMUM_ANDROID_GRADLE_PLUGIN_VER:
+        raise AndroidPostBuildError(f"Android gradle plugin versions below version {MINIMUM_ANDROID_GRADLE_PLUGIN_VER} is not supported.")
+    logger.info(f"Applying post-build for android gradle plugin version {gradle_version}")
+
+    # Validate the build directory exists
+    app_build_root = android_app_root_path / 'build'
+    if not app_build_root.is_dir():
+        raise AndroidPostBuildError(f"Android gradle build path: {app_build_root} is not a directory or does not exist.")
+
+    target_layout_root = android_app_root_path / 'src' / 'main' / 'assets'
+
+    if asset_mode == ASSET_MODE_LOOSE:
+
+        apply_loose_layout(project_root=project_root,
+                           target_layout_root=target_layout_root,
+                           android_app_root_path=android_app_root_path,
+                           native_build_folder=native_build_folder,
+                           build_config=build_config)
+
+    elif asset_mode == ASSET_MODE_PAK:
+
+        apply_pak_layout(project_root=project_root,
+                         target_layout_root=target_layout_root,
+                         asset_bundle_folder=asset_bundle_folder)
+
+
+if __name__ == '__main__':
+
+    try:
+        parser = argparse.ArgumentParser(description="Android post Gradle build step handler")
+        parser.add_argument('android_app_root', type=str, help="The base of the 'app' in the O3DE generated gradle script.")
+        parser.add_argument('--native-build-folder',  type=str, help="The native builder intermediate folder (for AGP version 7.x and newer)",
+                            default='o3de')
+        parser.add_argument('--build-config', type=str, help="The build configuration for the native build.", required=True, choices=SUPPORTED_BUILD_CONFIGS)
+        parser.add_argument('--project-root', type=str, help="The project root.", required=True)
+        parser.add_argument('--gradle-version', type=str, help="The version of Gradle.", required=True)
+        parser.add_argument('--asset-mode', type=str, help="The asset mode of deployment", default=ASSET_MODE_LOOSE, choices=SUPPORTED_ASSET_MODES)
+        parser.add_argument('--asset-bundle-folder', type=str, help="The sub folder from the project root where the pak files are located (For Pak Asset Mode)",
+                            default="AssetBundling/Bundles")
+
+        args = parser.parse_args(sys.argv[1:])
+
+        result_code = post_build_action(android_app_root=Path(args.android_app_root),
```

Comment:
`post_build_action` doesn't appear to return a value .. I guess that means it returns `0`?

Reply:
It shouldn't need a return, we use exceptions to propagate errors. Will remove